components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -259,11 +259,28 @@ private Response handleErrorResponse(OAuth2AccessTokenRespDTO oauth2AccessTokenR
             return handleServerError();
         } else {
             // Otherwise send back HTTP 400 Status Code
-            OAuthResponse response = OAuthASResponse
+            OAuthResponse.OAuthErrorResponseBuilder oAuthErrorResponseBuilder = OAuthASResponse
                     .errorResponse(HttpServletResponse.SC_BAD_REQUEST)
                     .setError(oauth2AccessTokenResp.getErrorCode())
-                    .setErrorDescription(oauth2AccessTokenResp.getErrorMsg())
-                    .buildJSONMessage();
+                    .setErrorDescription(oauth2AccessTokenResp.getErrorMsg());
+
+            try {
+                if (MapUtils.isNotEmpty(oauth2AccessTokenResp.getErrorParameterMap())) {
+                    for (Map.Entry<String, Object> entry: oauth2AccessTokenResp.getErrorParameterMap().entrySet()) {
+                        if (entry.getValue() instanceof String) {
```

Comment:
we are doing the same action in both if and else

Reply:
Do we need a branch?

Reply:
Fixed. 

For APIM 4.2.0 release, we will be using 6.9.x branch.